src/expr/symbol_table.cpp

```diff
@@ -205,29 +229,62 @@ bool OverloadedTypeTrie::markOverloaded(const string& name, Expr obj) {
 
   // types can be identical but vary on the kind of the type, thus we must
   // distinguish based on this
```

Comment:
Can you expand the comment here to explain a bit more about what is happening in the code below - it's a bit tricky.

Reply:
I've improved it in the latest commit